### PR TITLE
fix(braintree-web): add missing propertes to factory methods

### LINKED ISSUE #43346: [@types/braintree-web] hosted fields and 3d secure create methods are missing the authorization options
@types/braintree-web

According to the braintree js code hosted fields and secure 3d methods can be used by passing client or authorization token directly:

hosted fields:
- https://github.com/braintree/braintree-web/blob/5556f8e8c6ade6d0a1b5317e4fdae2700c09a8a8/src/hosted-fields/index.js#L109
- https://github.com/braintree/braintree-web/blob/5556f8e8c6ade6d0a1b5317e4fdae2700c09a8a8/src/hosted-fields/index.js#L257

three d secure:
- https://github.com/braintree/braintree-web/blob/master/src/three-d-secure/index.js#L27
- https://github.com/braintree/braintree-web/blob/5556f8e8c6ade6d0a1b5317e4fdae2700c09a8a8/src/three-d-secure/index.js#L128

@guy-shahine

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- `authorization`
- `version`

https://braintree.github.io/braintree-web/current/module-braintree-web_hosted-fields.html#.create
https://braintree.github.io/braintree-web/current/module-braintree-web_three-d-secure.html#.create

/cc @jbojcic1

Thanks!

Fixes: #43346

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

File: types/braintree-web/index.d.ts [hunk 1]
```diff
@@ -685,8 +686,8 @@ declare namespace braintree {
      *   }
      * }, callback);
      */
-     create(options: { client: Client, fields: HostedFieldFieldOptions, styles?: any }): Promise<HostedFields>;
-     create(options: { client: Client, fields: HostedFieldFieldOptions, styles?: any }, callback: callback): void;
+     create(options: { client: Client, authorization?: string, fields: HostedFieldFieldOptions, styles?: any }): Promise<HostedFields>;
+     create(options: { client: Client, authorization?: string, fields: HostedFieldFieldOptions, styles?: any }, callback: callback): void;
```
Comment (jbojcic1) [inline types/braintree-web/index.d.ts]: client should be optional. quote: 

> options.authorization A tokenizationKey or clientToken. Can be used in place of `options.client`.

Comment (jbojcic1) [reply types/braintree-web/index.d.ts]: in this braintree example you can how both secure 3 d and hosted fields can be used by specifying only authorization without the client https://codepen.io/braintree/pen/ezvymm

Comment (peterblazejewicz) [reply types/braintree-web/index.d.ts]: @jbojcic1 thx for the review, amended!
I obviously missed this in the docs I've linked myself 🤦‍♂

File: types/braintree-web/index.d.ts [hunk 2]
```diff
@@ -1165,8 +1166,8 @@ declare namespace braintree {
      */
     create(options: { client: Client }): Promise<ThreeDSecure>;
     create(options: { client: Client }, callback: callback): void;
-    create(options: { version: number, client: Client }): Promise<ThreeDSecure>;
-    create(options: { version: number, client: Client }, callback: callback): void;
+    create(options: { version?: 1 | '1' | 2 | '2' | '2-bootstrap3-modal' | '2-inline-iframe', client: Client }): Promise<ThreeDSecure>;
+    create(options: { version?: 1 | '1' | 2 | '2' | '2-bootstrap3-modal' | '2-inline-iframe', client: Client }, callback: callback): void;
```
Comment (jbojcic1) [inline types/braintree-web/index.d.ts]: @peterblazejewicz can you pls also make client optional and add authorization for 3d secure create too?

Comment (peterblazejewicz) [reply types/braintree-web/index.d.ts]: done, again, thanks for poiting the problem!

File: types/braintree-web/index.d.ts [hunk 3]
```diff
@@ -1165,8 +1166,8 @@ declare namespace braintree {
      */
     create(options: { client: Client }): Promise<ThreeDSecure>;
     create(options: { client: Client }, callback: callback): void;
-    create(options: { version: number, client: Client }): Promise<ThreeDSecure>;
-    create(options: { version: number, client: Client }, callback: callback): void;
+    create(options: { version?: 1 | '1' | 2 | '2' | '2-bootstrap3-modal' | '2-inline-iframe', client?: Client }): Promise<ThreeDSecure>;
```
Comment (jbojcic1) [inline types/braintree-web/index.d.ts]: still missing authorization param here

Comment (peterblazejewicz) [reply types/braintree-web/index.d.ts]: added with update, sorry to keep you busy :)

**Pull request merged**